$NetBSD: patch-bb,v 1.2 2019/10/11 09:08:35 jperkin Exp $

Do use non-standard getcwd(NULL, 0); This unbreaks cvs on Interix.
Disable broken custom getcwd() on Apple, it breaks on Catalina.

--- lib/xgetcwd.c.orig	2005-09-19 21:12:25.000000000 +0000
+++ lib/xgetcwd.c
@@ -25,8 +25,11 @@
 #include "xgetcwd.h"
 
 #include <errno.h>
+#include <limits.h>
 
+#ifndef __APPLE__
 #include "getcwd.h"
+#endif
 #include "xalloc.h"
 
 /* Return the current directory, newly allocated.
@@ -36,8 +39,8 @@
 char *
 xgetcwd (void)
 {
-  char *cwd = getcwd (NULL, 0);
+  char *cwd = xmalloc (PATH_MAX);
   if (! cwd && errno == ENOMEM)
     xalloc_die ();
-  return cwd;
+  return getcwd (cwd, PATH_MAX);
 }
